// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTAsymmetricKey";
option optimize_for = LITE_RUNTIME;

import public "Ciphertext.proto";
import public "Enums.proto";
import public "HDPath.proto";

message AsymmetricKey {
    optional uint32 version = 1;
    optional AsymmetricKeyType type = 2;
    optional KeyMode mode = 3;
    optional KeyRole role = 4;
    optional bytes key = 5;
    optional Ciphertext encryptedKey = 6;
    optional Ciphertext chaincode = 7;
    optional HDPath path = 8;
    optional uint32 bip32_parent = 9;
    optional bytes params = 10;
}
